graphite/graphite.go

```diff
@@ -43,8 +44,8 @@ func (r *Request) Query(host string) (Response, error) {
 		v.Add("until", fmt.Sprint(r.End.Unix()))
 	}
 	u := url.URL{
-		Scheme:   "http",
-		Host:     host,
+    Scheme:   strings.Split(host, ":")[0],
```

Comment:
ok then ,in this case instead of building a series on nested ifs to make sure the splits are working we could just use 
{code}
u1, err:= url.Parse(host)
u := url.URL{
  Scheme: u1.Scheme 
  Host: u1.Host 
{code}
etc 
Makes sense @mjibson  and @captncraig ?
MM